### ISSUE TITLE
Index template doesn't display comments or the comment form on the Comments block

### ISSUE THREAD

**Issue by happychait** [Automattic/wp-calypso#63847]
### Quick summary

The Index template doesn't show comments or comment form on the site, despite the editor preview showing the comment form.

### Steps to reproduce

1. Add Comments block to the `Index` template.
2. The comments form appears in the Site Editor.
3. Update the `Index` template in the Site Editor.

### What you expected to happen

Home page of the site (set to display latest posts) should show the comments, pagination and the comment form.

### What actually happened

Home page appears without any comments, comment form or anything related to comments.

### Context

Tested with this post: https://fullsiteeditor.blog/2022/02/15/pdf-embed-test/ which has two comments on it.

Site Editor shows at least the Comment form (although it won't show comments!) as in this screenshot: 

![Comments block in the Site Editor](https://user-images.githubusercontent.com/1519466/169453608-732eecd7-ab74-475d-ade7-916f2bd6a5bb.png)

But the home page of the site doesn't show anything related to comments as in this screenshot: 
![No comments on the home page](https://user-images.githubusercontent.com/1519466/169453634-370e4a80-86ae-48cf-9f95-9038553f7336.png)

Customer report: 35541399-hc followed up in 5226319-zen

**Comment by druesome** [comment]
Also reported in 5228548-zen.  User added the Comments block to a page, enabled comments, but form does not appear.

**Comment by cuemarie** [comment]
📌 HOUSEKEEPING
- Labels ✅ 
- Priority ✅ 
- Replicable on Core - No

📌 SCRUBBING
- Tested on Simple ✅ 
- Tested on AT ✅ 
- Tested on Self-hosted ✅ 

📌 FINDINGS/SCREENSHOTS/VIDEO
- I tried replicating this on Simple, AT, and Self-Hosted sites with Quadrat White (same as OP's user's theme), and I was able to see the comment form in all three environments on the index template.

<img width="600" alt="Screen Shot 2022-06-07 at 9 39 11 AM" src="https://user-images.githubusercontent.com/27249804/172436363-ac6b118f-d580-4a03-9e86-c8acf192e361.png">

<img width="600" alt="Screen Shot 2022-06-07 at 9 39 20 AM" src="https://user-images.githubusercontent.com/27249804/172436383-00766f3c-dc69-4eec-85bd-66642d103191.png">

<img width="600" alt="Screen Shot 2022-06-07 at 9 39 41 AM" src="https://user-images.githubusercontent.com/27249804/172436434-ab1598a1-c74c-43be-b132-13e5ee5b27b8.png">

📌 ACTIONS
- Closed as could not replicate, please reopen this issue if it reoccurs

📌 Message to Author
 - @happychait I was not able to replicate this issue on the Index template. Let me know if my testing steps missed any steps that you took, and feel free to reopen if you believe this needs further testing.

**Comment by happychait** [comment]
Thanks for testing this, @cuemarie! This is not a bug. 

Here is what happened:

When I reported this issue, I tested with the  **Twenty Twenty-Two (Red)** theme which has the `Home` template as well as `Index` template. I added the `Comments` block to the `Index` template, but didn't realize that `Home` template takes precedence, when it exists, over the `Index` template. So what was rendering the home page at the time I report this non-bug was the `Home` template that didn't have the `Comments` block in it, and so the comments and the comment form are not displayed on the home page of my site. That's how I missed this!

I just tested again by adding the `Comments` block to the `Home` template, and can see the comments on the home page.


Sorry for the trouble and extra work to scrub this non-bug.
